_posts/jasonwang/2016-04-29-jasonwang-finalproject.md

```diff
@@ -0,0 +1,51 @@
+
+I then began data analysis. While I did have an idea on what I wanted to do, it was not until the beginning of data analysis where I was able to pinpoint on what results I wanted in the end. While I initially said that I wanted to compare certain values together, I thought about the meaning of such comparisons, and realized that overall, such outputs would not teach anyone much about data trends. However, general distributions, such as the average rating over time, or the average comment number over time, gives a picture of the Youtube landscape at the time. 
+
+I still took a modular approach, although a slightly different modular approach. I first set up a test sequence- where data flows freely between demarcated sections and produces the results. Once that was completed, I split the sections into separate functions (but still without data requesting). I really did think that the UI would be pretty easy to implement- but between the differences of the while loop and minimizing the data that flows between functions, the functions still had to be restructured (and new ones made). Perhaps in this case, I should have also made a rough UI version so that the functions that I had made were also designed to work with the While loops.
+
+Between this project and the last, I personally believe that my debugging skill had increased dramatically. While in the last project I had to keep playing around with values (and sometimes breaking it more) to try and isolate why certain functions weren’t working as intended, zeroing in on problems in this project was much easier. This may be helped by the fact that you can track the data as it flows through, but I also think that my deeper understanding of Python since the last project allowed me to stop and think, “Why?”. And from “why”, we can narrow down the possible areas of interest.
+
+Honestly, between the structure of this project and the last project, I thoroughly appreciated the open-ended, milestone-focused aspect of keeping on track. The milestones really did help me quickly narrow down what topics I wanted to than if it was completely free, and the updates pushed me to make progress between each class. While I’m not saying that I would be planning on finishing the whole project 30 minutes before, the milestones gave me a good plan to plan my progress around, and plan what I would try to accomplish that day. The permanent groups was also pretty helpful. Being able to keep up with certain people over time gave me a better understanding and appreciation of what they were doing, and gave me some insights that I eventually used in the construction of my program.
```

Comment:
The benefits of using project management!  I imposed these requirements on you but there's no reason you couldn't impose them on youself in future projects, and I hope you do.